aderyn_core/src/detect/mod.rs

```diff
@@ -24,4 +24,74 @@ macro_rules! capture {
     };
 }
 
+#[macro_export]
+macro_rules! issue_detector {
+    (
+        $detector_struct:ident;
+
+        severity: $detector_severity:ident,
+        title: $detector_title:tt,
+        desc: $detector_desc:tt,
+        name: $detector_name:ident,
+
+        |$context: ident| $e:expr
+    ) => {
+
+        #[derive(Default)]
+        pub struct $detector_struct {
+            found_instances: std::collections::BTreeMap<(String, usize, String), NodeID>,
+        }
+
+        impl $crate::detect::detector::IssueDetector for $detector_struct {
+
+            fn detect(&mut self, context: &$crate::context::workspace_context::WorkspaceContext) -> Result<bool, Box<dyn std::error::Error>> {
+
+                let $context = context;
+
+                macro_rules! grab {
```

Comment:
Why `grab` and not `capture`?

Reply:
[Macro Hygiene Rule](https://veykril.github.io/tlborm/syntax-extensions/hygiene.html) in Rust makes it such that,  inside a macro, an externally defined macro can't make a reference to "self". Here, `capture!()` is defined external to the `issue_detector!` so it won't be able to refer to it. That makes it "unhygenic". Therefore, had to define one inside :)